src/Layer/Layer.js

```diff
@@ -0,0 +1,167 @@
+     * layerToChange.opacity = 0.5;
+     * view.notifyChange(); // update viewer
+     *
+     * // Listen properties
+     * const layerToListen = view.getLayers(layer => layer.id == 'idLayerToListen')[0];
+     * layerToListen.addEventListener('visible-property-changed', (event) => console.log(event));
+     * layerToListen.addEventListener('opacity-property-changed', (event) => console.log(event));
+     * @constructor
+     * @protected
+     * @param {Object} config
+     */
+    constructor(config) {
+        super();
+
+        Object.assign(this, config);
+
+        Object.defineProperty(this, 'id', {
+            value: config.id,
+            writable: false,
+        });
+
+        Object.defineProperty(this, 'type', {
+            value: config.type,
+            writable: false,
+        });
+
+        // Default properties
+        this.options = config.options || {};
+
+        if (!this.updateStrategy) {
+            this.updateStrategy = {
+                type: STRATEGY_MIN_NETWORK_TRAFFIC,
+            };
+        }
+
+        // TODO remove this warning and fallback after the release following v2.3.0
+        if (!this.format && this.options.mimetype) {
+            console.warn('layer.options.mimetype is deprecated, please use layer.format');
+            this.format = this.options.mimetype;
+        }
+
+        this.defineLayerProperty('frozen', false);
+    }
+
+    defineLayerProperty(propertyName, defaultValue, onChange) {
+        const existing = Object.getOwnPropertyDescriptor(this, propertyName);
+        if (!existing || !existing.set) {
+            let property = this[propertyName] == undefined ? defaultValue : this[propertyName];
+
+            Object.defineProperty(
+                this,
+                propertyName,
+                {
+                    get: () => property,
+                    set: (newValue) => {
+                        if (property !== newValue) {
+                            const event = { type: `${propertyName}-property-changed`, previous: {}, new: {} };
+                            event.previous[propertyName] = property;
+                            event.new[propertyName] = newValue;
+                            property = newValue;
+                            if (onChange) {
+                                onChange(this, propertyName);
+                            }
+                            this.dispatchEvent(event);
+                        }
+                    },
+                });
+        }
+    }
+}
+
+export default Layer;
+
+export const ImageryLayers = {
+    // move this to new index
+    // After the modification :
+    //      * the minimum sequence will always be 0
+    //      * the maximum sequence will always be layers.lenght - 1
+    // the ordering of all layers (Except that specified) doesn't change
+    moveLayerToIndex: function moveLayerToIndex(layer, newIndex, imageryLayers) {
```

Comment:
maybe this should be in `geometryLayer` because the color and elevation layers are only attached to geometric layers : `geometryLayer.moveColorLayerToIndex(layer, index)`?

Reply:
This kind of makes sense, but this method is used in `ColorLayersOrdering` too; how do you see the integration with this one, very specific to color layers ?